plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/ScanQueryPageSource.java

```diff
@@ -158,8 +158,23 @@ public Page getNextPage()
             Map<String, Object> document = hit.getSourceAsMap();
 
             for (int i = 0; i < decoders.size(); i++) {
-                String field = columns.get(i).name();
-                decoders.get(i).decode(hit, () -> getField(document, field), columnBuilders[i]);
+                OpenSearchColumnHandle columnHandle = columns.get(i);
+                if (!columnHandle.dereferencePath().isEmpty()) {
+                    Map<String, Object> value = (Map<String, Object>) getField(document, columnHandle.baseName());
+                    if (value != null) {
+                        for (int j = 0; j < columnHandle.dereferencePath().size() - 1; j++) {
+                            value = (Map<String, Object>) getField(value, columnHandle.dereferencePath().get(j));
+                            if (value == null) {
+                                break;
+                            }
+                        }
+                    }
+                    Map<String, Object> leafDocument = value;
```

Comment:
This variable seems unnecessary.

Reply:
lambdas require effectively final variable.